internal/environment/env.go

```diff
@@ -32,14 +32,28 @@ type CI struct {
 	getenv GetEnv
 }
 
+// interface to allow dependency injection that satisfies contract
+type OutputI interface {
```

Comment:
nit: just wondering if calling the interface `OutputI` is idiomatic.

Reply:
At the time, I would have liked to stay closer to the best practice of "Accept interfaces, return structs".

Since environment package SetOutput accepts `map[string]environment.OutputI`. It's not the same as `map[string]command.OutputI` even if the value satisfies the interface constraints.

I resorted to making a custom exported type of `type OutputMap map[string]OutputI`

Reply:
I think we can make this more idiomatic my reverting back the old interface methods I removed from the environment package: `AddOutput(...)`

Reply:
My comment was just about the naming of OutputI  with the suffix I  🙂 
I like the idea of "Accept interfaces, return structs"

Reply:
Apologies! I've seen the naming convention employed but can't recall.

Reply:
It's more common in Go to have an `*er` interface. eg: `io.Reader` or `io.Writer`.

I think I've seen `*I` or `*Impl` as a Java convention.

Reply:
@ggambetti I'll adopt *er naming convention for interfaces 👍